image-customize

```diff
@@ -82,6 +83,114 @@ class InstallAction(ActionBase):
         machine_instance.execute(f"{install_command} {package}", timeout=1800)
 
 
+class BuildAction(ActionBase):
+    '''Build and install distribution package(s) from dist tarball'''
+
+    quick = False
+    verbose = False
+
+    @staticmethod
+    def execute(machine_instance, tarball):
+        # upload the tarball
+        tarname = os.path.basename(tarball)
+        vm_tar = os.path.join("/var/tmp", tarname)
+        # looks like cockpit-foo-123.10.gdbde381.tar.xz
+        m = re.search(r'([^-]+)\.tar(\..*)?$', tarname)
+        if not m:
+            raise ValueError(f"cannot parse version number from tarball {tarname}")
+        version = m.group(1)
+        machine_instance.upload([tarball], vm_tar, relative_dir=".")
+
+        machine_instance.execute("rm -rf /var/tmp/build; mkdir -p /var/tmp/build")
```

Comment:
Why don't you deal with that from the scripts?

Reply:
Mostly DRY - because that's common to all backends. If you prefer, I can move it into the backend functions, I don't have a strong opinion about it.

Reply:
Nah.  Fair enough :)

Reply:
Can't -- Debian doesn't have a "builder" system user, pbuilder does that internally.

Reply:
No, I mean the opposite: that it should be created at the point where we assign the permissions (or not), rather than creating it and assigning the permissions later.

Reply:
Sure, the prep and build could happen in a `su` subshell, and we could triplicate the build directory creation. Given how little this chown thing matters in practice (i.e. "not at all"), I prefer this over adding yet another nesting level to these commands. It's how we use our test VMs (root shells). If rewriting it is what it takes to land this PR, I'll do it, but I really don't like making the code more complicated..

Reply:
Ugh, sorry -- my last sentence was unnecessarily snarky. Please accept my apology! I didn't mean that personally, just that I'd rather like to discuss that again -- for such glue stuff, simplicity and copy&paste-ability are desirable IMHO.